### ISSUE TITLE
carrier install --system-domain does not accept k3d.localhost

### ISSUE THREAD

**Issue by troytop** [SUSE/carrier#84]
```
$ carrier install --system-domain k3d.localhost
🚢 Carrier installing...

Configuration...
  🧭 system_domain: 172.23.0.2.nip.io

⚠️  Traefik Ingress already installed, skipping
```

Have not tried with other "normal" domains. 

The `-i` interactive option worked, and ingresses were created with the `k3d.localhost` domain.

**Comment by andreas-kupries** [comment]
Adding a bit more output, i.e. additional calls to `c.showInstallConfiguration(deployments)` I found that

  - The config is ok after `kubernetes.NewCLIOptionsReader(cmd)`. That is the function setting config per cli flags, like `--system-domain`.
  - It is bad after `kubernetes.NewDefaultOptionsReader()` was run.

This is not necessarily an issue with the latter. It may be an issue in the former not signaling correctly that its value should be kept.

:eyes: on these now.

**Comment by andreas-kupries** [comment]
Ok. It was in the default reader.

In the beginning the defaults were set first, and then the other readers had their turn. That made it ok to set the values unconditionally.

At some point, which I have not bothered to determine right now, the order changed, defaults became the last thing to be set, after the other readers. At that point it was forgotten to make the default reader conditional, i.e. to skip all the options which now had user-specified information set into them.

This has been fixed now, with PR #85.
